Automattic/jetpack
# Sync: initialize Dedicated Sync endpoint earlier

## Proposed changes:
Initialize dedicated sync endpoint earlier in the `init` hook to avoid cookie conflicts.

### Other information:


## Jetpack product discussion
p1686285707645099-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
Make sure the code makes sense.

Do not checkout the code just yet, the issue may be a bit hard to reproduce.
1. Create a JN site with Woo, WCPay, Jetpack, and Smooth Generator.
2. Connect Jetpack, generated a few products.
3. Go to "WooCommerce -> Settings -> Accounts & Privacy" and enable "Allow customers to log into an existing account during checkout"
4. Create a new user.
5. In a private browser window, add a few products to the cart, go to checkout and login right there.
6. The products will probably still be in the cart as expected. Go to Cart and remove them. Then add a few products to the cart manually and remove them again. Make sure your cart is empty.
7. Close the window to clean the session, and repeat 6-7 a few times, and eventually you should lose the products in your card after logging in.
8. Checkout the PR and try a few more times, confirm the issue got fixed.
